CoreScriptsRoot/Modules/VR/Panel3D.lua

```diff
@@ -129,6 +141,40 @@ function Panel3D.SetModalPanel(panel)
 	lastModal = currentModal
 	currentModal = panel
 end
+
+function Panel3D.RaycastOntoPanel(part, parentGui, gui, ray)
+	local partSize = part.Size
+	local partThickness = partSize.Z
+	local partWidth = partSize.X
+	local partHeight = partSize.Y
+
+	local planeCF = part:GetRenderCFrame()
+	local planeNormal = planeCF.lookVector
+	local pointOnPlane = planeCF.p + (planeNormal * partThickness * 0.5)
+
+	local worldIntersectPoint = Util.RayPlaneIntersection(ray, planeNormal, pointOnPlane)
+	if worldIntersectPoint then
+		local parentGuiWidth, parentGuiHeight = parentGui.AbsoluteSize.X, parentGui.AbsoluteSize.Y
+		local localIntersectPoint = (planeCF:pointToObjectSpace(worldIntersectPoint) / currentHeadScale) * Vector3.new(-1, 1, 1) + Vector3.new(partWidth / 2, -partHeight / 2, 0)
+		local lookAtPixel = Vector2.new((localIntersectPoint.X / partWidth) * parentGuiWidth, (localIntersectPoint.Y / partHeight) * -parentGuiHeight)
+		
+		--fire mouse enter/leave events if necessary
```

Comment:
We did, but this covers the entire panel. I'll have to check if we actually use it still.